src/transforms/bin.js

```diff
@@ -98,6 +98,10 @@ function binn(
   // Compute the outputs.
   outputs = maybeBinOutputs(outputs, inputs);
   reduceData = maybeBinReduce(reduceData, identity);
+  if (typeof sort === "string" && sort.startsWith("-")) {
+    sort = sort.slice(1);
+    reverse = !reverse;
+  }
```

Comment:
A subtle gotcha here: `sort: "-x"` doesn’t mean “sort by *x* in natural ascending order and then reverse the resulting order”; it means “sort by *x* in natural descending order”. The difference is that we want nulls/unordered values to be last when using *-x*, whereas reversing after sorting would put them first.

The ChannelDomainOptions interface is confusing in this regard because it calls the option **reverse**, but it really means whether to use ascending or descending order (per the documentation):

https://github.com/observablehq/plot/blob/0490ba6f1dba3403ed050e453e9c0839ecf635a5/src/channel.d.ts#L195-L196

I commented on this indirectly in #1460, and I’m going to add some context to that issue now to clarify the confusion.

In any case, here we ideally don’t want to flip the **reverse** option; we want to use descending rather than ascending natural order. That’s why I added the reverseOrder helper for the basic sort transform:

https://github.com/observablehq/plot/blob/bef0ec98085669f0eb1a979af8bdaf534087aaad/src/transforms/basic.js#L143-L149

I think this might mean that the maybeSort helper should take an order argument, and that the bin and group transform should support `order: "descending"` and `order: "ascending"` as complements to the sort option, instead of only supporting a reverse option?